### PR TITLE
Fix/natspec

### PR DESCRIPTION
**Pull request by Andreadinenno**
Natspecs for `IERC777Sender` and `IERC777Recipient` 

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Does your submission pass tests?
* [x] Have you lint your code locally prior to submission?

### CODE REVIEW

File: contracts/IERC777Recipient.sol [hunk 1]
```diff
@@ -4,14 +4,28 @@
...
 interface IERC777Recipient {
+
     /**
-     * @dev Called by an `IERC777` token contract whenever tokens are being
-     * moved or created into a registered account (`to`).
+    * @notice Called by a `ERC777` token contract whenever tokens are being moved or created into a account (`to`).
+    * This call occurs after the contract state (i.e. balances) is updated and may revert to prevent transfer.
+    * It is MANDATORY for a Smart Contract that intends to receive and manage tokens to implement this interface, 
+    * or each transfer to it would revert.
+    * @param operator address of the operator, if any, that triggered the transfer
```
Comment (cmjc) [inline contracts/IERC777Recipient.sol:21]: Oh, we're missing the formatting for the params - operator -> `operator`

File: contracts/IERC777Sender.sol [hunk 2]
```diff
@@ -5,13 +5,26 @@
...
+*/
+
 interface IERC777Sender {
 
-    //this call occurs before the token state is updated  and may revert to prevent the operation
+    /**
+    * @notice Called by a `ERC777` token contract before tokens are being transferred or burned.
+    * This call occurs before the contract state (i.e. balances) is updated and may revert to prevent transfer.
+    * It is OPTIONAL both for a Smart Contract and a EOA to implement this interface.
+    * @param operator address of the operator, if any, that triggered the transfer
```
Comment (cmjc) [inline contracts/IERC777Sender.sol:20]: Param formatting: operator -> `operator`

File: docs/IAssetToken.md [hunk 3]
```diff
@@ -216,11 +216,17 @@ Will revert if the caller address is either the `owner`, `fundingDelegate`, `eme
...
 Transfer tokens between addresses.
 
-Will trigger `tokensToSend` and `tokensReceived` hooks as per ERC777 Will revert if `amount` doesn't respect the granularity.
+Will trigger `tokensToSend` and `tokensReceived` hooks as per ERC777.
+
+Will revert if `amount` doesn't respect the granularity.
+
+Will revert if the caller is not allowed (blacklisted or not yet whitelisted).
+
+Will revert if the contract is in frozen state after an `emergencyStop`.
```
Comment (cmjc) [inline docs/IAssetToken.md:225]: In _a_ frozen state.

File: docs/IAssetToken.md [hunk 4]
```diff
@@ -216,11 +216,17 @@ Will revert if the caller address is either the `owner`, `fundingDelegate`, `eme
...
 
-Will revert if the caller is not allowed (blacklisted or not yet whitelisted) Will revert if the contract is in frozen state after an `emergencyStop`.
+Will revert if the recipient is  either the `owner`, `fundingDelegate`, `emergencyDelegate`, `listController`.
```
Comment (cmjc) [inline docs/IAssetToken.md:227]: Hmm.

>either the `owner`, `fundingDelegate`, `emergencyDelegate`, `listController`

or could be:

>either the `owner`, `fundingDelegate`, `emergencyDelegate`, or `listController`

File: docs/IAssetToken.md [hunk 5]
```diff
@@ -236,9 +242,17 @@ Will trigger `tokensToSend` and `tokensReceived` hooks as per ERC777.
 
 Will revert if `amount` doesn't respect the granularity.
 
-Will revert if the caller is not an operator of the token holder Will revert if the operator or the token holder are not allowed (blacklisted or not yet whitelisted) Will revert if the contract is in frozen state after an `emergencyStop`.
+Will revert if the caller is not an operator of the token holder.
+
+Will revert if the operator or the token holder are not allowed (blacklisted or not yet whitelisted).
```
Comment (cmjc) [inline docs/IAssetToken.md:247]: >Will revert if the operator or the token holder are not allowed to use the token contract (i.e. blacklisted or not  whitelisted if the contract is initialised in a list mode of operation).

What do you think? Is that a useful clarification or unnecessary?

File: docs/IAssetToken.md [hunk 6]
```diff
@@ -236,9 +242,17 @@ Will trigger `tokensToSend` and `tokensReceived` hooks as per ERC777.
 
 Will revert if `amount` doesn't respect the granularity.
 
-Will revert if the caller is not an operator of the token holder Will revert if the operator or the token holder are not allowed (blacklisted or not yet whitelisted) Will revert if the contract is in frozen state after an `emergencyStop`.
+Will revert if the caller is not an operator of the token holder.
+
+Will revert if the operator or the token holder are not allowed (blacklisted or not yet whitelisted).
+
+Will revert if the contract is in frozen state after an `emergencyStop`.
```
Comment (cmjc) [inline docs/IAssetToken.md:249]: in a frozen state

File: docs/IAssetToken.md [hunk 7]
```diff
@@ -236,9 +242,17 @@ Will trigger `tokensToSend` and `tokensReceived` hooks as per ERC777.
...
 Will revert if `amount` doesn't respect the granularity.
 
-Will revert if the caller is not an operator of the token holder Will revert if the operator or the token holder are not allowed (blacklisted or not yet whitelisted) Will revert if the contract is in frozen state after an `emergencyStop`.
+Will revert if the caller is not an operator of the token holder.
+
+Will revert if the operator or the token holder are not allowed (blacklisted or not yet whitelisted).
+
+Will revert if the contract is in frozen state after an `emergencyStop`.
+
+Will revert if the recipient is  either the `owner`, `fundingDelegate`, `emergencyDelegate`, `listController`.
```
Comment (cmjc) [inline docs/IAssetToken.md:251]: or `listController`

File: docs/IERC777Recipient.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
+# [🔗](/interfaces/IERC777Recipient.sol#L7) IERC777Recipient
+**Author** _Andrea Di Nenno_
+
+This interface allows a token receiver (EOA or Smart Contract) to add extra logic to control his incoming tokens.
+
+
+# Functions
+## [🔗](/interfaces/IERC777Recipient.sol#L15) `tokensReceived(address operator, address from, address to, uint256 amount, bytes userData, bytes operatorData)`
+
+Called by a `ERC777` token contract whenever tokens are being moved or created into a registered account (`to`).
```
Comment (cmjc) [inline docs/IERC777Recipient.md:10]: an `ERC777`

File: docs/IERC777Sender.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
+# [🔗](/interfaces/IERC777Sender.sol#L7) IERC777Sender
+**Author** _Andrea Di Nenno_
+
+This interface allows a token sender (EOA or Smart Contract) to add extra logic to control his outgoing tokens.
+
+
+# Functions
+## [🔗](/interfaces/IERC777Sender.sol#L15) `tokensToSend(address operator, address from, address to, uint256 amount, bytes userData, bytes operatorData)`
+
+Called by a `ERC777` token contract before tokens are being transferred or burned.
```
Comment (cmjc) [inline docs/IERC777Sender.md:10]: an `ERC777`

Comment (cmjc) [reply docs/IERC777Sender.md:10]: Don't think we need the being:

>Called by a `ERC777` token contract before tokens are transferred or burned.

File: docs/IERC777Sender.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+
+# Functions
+## [🔗](/interfaces/IERC777Sender.sol#L15) `tokensToSend(address operator, address from, address to, uint256 amount, bytes userData, bytes operatorData)`
+
+Called by a `ERC777` token contract before tokens are being transferred or burned.
+
+This call occurs before the contract state (i.e. balances) is updated and may revert to prevent transfer.
+
+It is OPTIONAL both for a Smart Contract and a EOA to implement this interface.
```
Comment (cmjc) [inline docs/IERC777Sender.md:14]: an EOA

File: interfaces/IERC777Recipient.sol [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+// SPDX-License-Identifier: LGPL-3.0+
+
+pragma solidity ^0.5.0;
+
+/**
+* @title ERC777 Token Recipient Interface
+* @author Andrea Di Nenno
+* @notice This interface allows a token receiver (EOA or Smart Contract) to add extra logic 
+* to control his incoming tokens.
```
Comment (cmjc) [inline interfaces/IERC777Recipient.sol:11]: Is 'their' better than 'his' as it is gender agnostic?

File: interfaces/IERC777Recipient.sol [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+*/
+
+interface IERC777Recipient {
+
+    /**
+    * @notice Called by a `ERC777` token contract whenever tokens are being moved or created into a account (`to`).
```
Comment (cmjc) [inline interfaces/IERC777Recipient.sol:17]: Called by an `ERC777` token contract whenever tokens are being moved or created into an account

File: interfaces/IERC777Recipient.sol [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+    * This call occurs after the contract state (i.e. balances) is updated and may revert to prevent transfer.
+    * It is MANDATORY for a Smart Contract that intends to receive and manage tokens to implement this interface, 
+    * or each transfer to it would revert.
+    * @param operator address of the operator, if any, that triggered the transfer
```
Comment (cmjc) [inline interfaces/IERC777Recipient.sol:21]: param formatting - e.g. `operator`

File: interfaces/IERC777Sender.sol [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+
+// SPDX-License-Identifier: LGPL-3.0+
+
+pragma solidity ^0.5.0;
+
+/**
+* @title ERC777 Token Sender Interface
+* @author Andrea Di Nenno
+* @notice This interface allows a token sender (EOA or Smart Contract) to add extra logic 
+* to control his outgoing tokens.
```
Comment (cmjc) [inline interfaces/IERC777Sender.sol:11]: their outgoing tokens

File: interfaces/IERC777Sender.sol [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+*/
+
+interface IERC777Sender {
+
+    /**
+    * @notice Called by a `ERC777` token contract before tokens are being transferred or burned.
```
Comment (cmjc) [inline interfaces/IERC777Sender.sol:17]: Called by an `ERC777` token contract before tokens are transferred or burned.

Don't think the being is needed here.

File: interfaces/IERC777Sender.sol [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+    * This call occurs before the contract state (i.e. balances) is updated and may revert to prevent transfer.
+    * It is OPTIONAL both for a Smart Contract and a EOA to implement this interface.
+    * @param operator address of the operator, if any, that triggered the transfer
```
Comment (cmjc) [inline interfaces/IERC777Sender.sol:20]: param formatting e.g. `operator`

**Review by cmjc — CHANGES_REQUESTED** [review]
Found a few formatting points and some a/an's.

File: contracts/IERC777Recipient.sol (hunk 1 above)
Comment (Andreadinenno) [reply contracts/IERC777Recipient.sol:21]: that formatting is produced automatically with the @param keyword.

File: docs/IAssetToken.md (hunk 5 above)
Comment (Andreadinenno) [reply docs/IAssetToken.md:247]: Useful :)

File: contracts/IERC777Sender.sol [hunk 17]
```diff
@@ -5,13 +5,26 @@
...
+* @author Andrea Di Nenno
+* @notice This interface allows a token sender (EOA or Smart Contract) to add extra logic 
+* to control his outgoing tokens.
+*/
+
 interface IERC777Sender {
 
-    //this call occurs before the token state is updated  and may revert to prevent the operation
+    /**
+    * @notice Called by a `ERC777` token contract before tokens are being transferred or burned.
```
Comment (cmjc) [inline contracts/IERC777Sender.sol:17]: Called by an `ERC777` token contract before tokens are transferred or burned.

File: docs/IERC777Recipient.md [hunk 18]
```diff
@@ -0,0 +1,26 @@
+# [🔗](/interfaces/IERC777Recipient.sol#L7) IERC777Recipient
+**Author** _Andrea Di Nenno_
+
+This interface allows a token receiver (EOA or Smart Contract) to add extra logic to control its incoming tokens.
+
+
+# Functions
+## [🔗](/interfaces/IERC777Recipient.sol#L16) `tokensReceived(address operator, address from, address to, uint256 amount, bytes userData, bytes operatorData)`
+
+Called by an `ERC777` token contract whenever tokens are being moved or created into a account.
```
Comment (cmjc) [inline docs/IERC777Recipient.md:10]: Sorry missed the -ing and a-an

Called by an `ERC777` token contract whenever tokens are moved or created into an account.

**Review by cmjc — CHANGES_REQUESTED** [review]
I found 2 more minor things, sorry 😬

**Review by cmjc — APPROVED** [review]
lgtm

**Pull request merged**